cabal-install/src/Distribution/Client/Dependency.hs

```diff
@@ -756,14 +753,8 @@ standardInstallPolicy installedPkgIndex sourcePkgDb pkgSpecifiers =
 
 -- ------------------------------------------------------------
 
-chooseSolver :: Verbosity -> PreSolver -> CompilerInfo -> IO Solver
-chooseSolver _verbosity preSolver _cinfo =
-  case preSolver of
-    AlwaysModular -> do
-      return Modular
-
-runSolver :: Solver -> SolverConfig -> DependencyResolver UnresolvedPkgLoc
-runSolver Modular = modularResolver
+runSolver :: SolverConfig -> DependencyResolver UnresolvedPkgLoc
+runSolver = modularResolver
```

Comment:
Can we rename `modularSolver` to `runSolver` in `Distribution.Solver.Module`? I see this line as a left-over.

Reply:
I think that removing the name "modular" from the existing dependency solver is a bigger change and should be done in a separate PR. I agree that this function should be removed, though. `resolveDependencies` could call `modularResolver` directly.

Reply:
I am happy with that.

Reply:
I splited the PR, here are the changes that do NOT modify cabal behavior https://github.com/haskell/cabal/pull/9282!